business/istio_status.go

```diff
@@ -89,38 +95,73 @@ func (iss *IstioStatusService) getIstioComponentStatus() (IstioComponentStatus,
 	return deploymentStatus.merge(istiodStatus), nil
 }
 
-func (iss *IstioStatusService) getComponentNamespacesWorkloads() ([]apps_v1.Deployment, error) {
+func (iss *IstioStatusService) getComponentNamespacesWorkloads() ([]workloadInfo, error) {
 	var wg sync.WaitGroup
 
 	nss := map[string]bool{}
-	deps := make([]apps_v1.Deployment, 0)
+	deps := make([]workloadInfo, 0)
 
 	comNs := getComponentNamespaces()
 
-	depsChan := make(chan []apps_v1.Deployment, len(comNs))
-	errChan := make(chan error, len(comNs))
+	depsChan := make(chan []workloadInfo, len(comNs)*2)
+	errChan := make(chan error, len(comNs)*2)
 
 	for _, n := range comNs {
 		if !nss[n] {
-			wg.Add(1)
+			wg.Add(2)
```

Comment:
I'm thinking on this logic.
And the same problem we have here (to use a DaemonSet vs a Deployment) is going likely to happen with other workload type.

So I wonder if this logic should just pull "workloads" from the business layer; then, we will solve the scenario where the istio workload is from one type or another, and we won't duplicate a similar logic that is keep synced in the workload package.

Or perhaps 99% will be always a Deployment, and a 1% a cornercase like a DaemonSet, but if we have another similar personalization, we'll have same problem again (thinking on Job controller or a ReplicationController, or another controller different than Deployment).

I don't know, I'll keep studying these logic.

As it happens in other area, as the Graph Appenders, I think that when using workloads, they should use de business package instead of query directly.

Reply:
Another comment around this, it happens also, that this doesn't update the DaemonSet controller view in the Workload level.

(DaemonSets and Jobs were "cornercases" for Istio when we develop this logic a.k.a. there were not even supported in Istio in first versions, so, for that case when a DaemonSet/Job was present we tried to resolve from Pods instead from the controllers. That has some known limitations, like non visibility of these controllers scaled to 0 for example, or the lack of fetching info from the controller itself).